src/core_atmosphere/physics/mpas_atmphys_driver_sfclayer.F

```diff
@@ -184,14 +192,18 @@ subroutine allocate_sfclayer(configs)
 
  sfclayer_select: select case (trim(sfclayer_scheme))
 
-    case("sf_monin_obukhov")
+    case("sf_monin_obukhov","sf_monin_obukhov_rev")
        if(.not.allocated(fh_p)) allocate(fh_p(ims:ime,jms:jme))
        if(.not.allocated(fm_p)) allocate(fm_p(ims:ime,jms:jme))
        if(config_frac_seaice) then
           if(.not.allocated(fh_sea)) allocate(fh_sea(ims:ime,jms:jme))
           if(.not.allocated(fm_sea)) allocate(fm_sea(ims:ime,jms:jme))
        endif
 
+       if(.not.allocated(waterdepth_p)) allocate(waterdepth_p(ims:ime,jms:jme))
+       if(.not.allocated(lakedepth_p) ) allocate(lakedepth_p(ims:ime,jms:jme) )
+       if(.not.allocated(lakemask_p)  ) allocate(lakemask_p(ims:ime,jms:jme)  )
```

Comment:
Just to note that the sf_monin_obukhov case do not require these three fields. But I'm ok with what you have here.

Reply:
Thanks for the suggestions. I modified subroutines allocate_sfclayer and deallocate_sfclayer. Arrays waterdepth_p, lakedepth_p, and lakemask_p are now allocated and deallocated only for the option sf_monin_obukhov_rev.

Reply:
I thought this would make the conditionals too complicated. But thanks for the change.